test/widget_tests/screens/recipe_feed_screen_test.dart

```diff
@@ -0,0 +1,205 @@
+import 'package:mobile_app/src/widgets/logo.dart';
+import 'package:mobile_app/src/widgets/recipe_feed/recipe_card_mobile.dart';
+import 'package:mobile_app/src/widgets/recipe_feed/recipe_card_tablet.dart';
+import 'package:mobile_app/src/widgets/top_bar_tablet.dart';
+
+import 'package:mockito/mockito.dart';
+import '../../cloud_firestore_mock.dart';
+
+class MockPostServices extends Mock implements PostServices {}
+
+class MockUserServices extends Mock implements UserServices {}
+
+void main() {
+  late PostBloc postBloc;
+  late MockPostServices mockPostServices;
+  late MockUserServices mockUserServices;
+  setUpAll(() async {
+    setupCloudFirestoreMocks();
+    Firebase.initializeApp();
+    HttpOverrides.global = null;
+  });
+  setUp(() {
+    mockPostServices = MockPostServices();
+    mockUserServices = MockUserServices();
+    postBloc = PostBloc(
+        postServices: mockPostServices, userServices: mockUserServices);
+  });
+
+  final mobileWidget = ScreenUtilInit(
+      designSize: Size(375, 812),
+      builder: () => MaterialApp(
+            home: BlocProvider(
+              create: (context) => postBloc,
+              child: Scaffold(
+                body: RecipeFeedScreen(),
+              ),
+            ),
+          ));
+
+  final tabletWidget = ScreenUtilInit(
+      designSize: Size(748, 1024),
+      builder: () => MaterialApp(
+              home: BlocProvider(
+            create: (context) => postBloc,
+            child: Scaffold(
+              body: RecipeFeedScreen(),
+            ),
+          )));
+  group('mobile test', () {
+    testWidgets('Should render Recipe Feed Screen Mobile widgets',
+        (WidgetTester tester) async {
+      await tester.pumpWidget(mobileWidget);
+      final logoFinder = find.byType(Logo);
+      expect(logoFinder, findsOneWidget);
+
+      final buttonFinder = find.byType(IconButtonCustom);
+      expect(buttonFinder, findsNWidgets(2));
+
+      expect(find.byType(PreferredSize), findsOneWidget);
+    });
+    testWidgets('renders RecipeCardMobile when state is PostLoadSuccess',
+        (WidgetTester tester) async {
+      await tester.pumpWidget(mobileWidget);
+      final StatefulElement _recipeFeedScreenElement =
+          tester.element(find.byType(RecipeFeedScreen));
+      final RecipeFeedScreenState _recipeFeedScreenState =
+          _recipeFeedScreenElement.state as RecipeFeedScreenState;
+      final pageViewFinder = find.byType(PageView);
+      final mockPost = Post(
+          backgroundImage:
+              "https://img.hoidap247.com/picture/question/20200718/large_1595063159202.jpg",
+          comments: 1,
+          likes: 1,
+          name: 'hy',
+          time: 111,
+          recipeId: 'ádasd',
+          userId: 'vip',
+          description: 'hgildasgi');
+      final mockUser = UserModel(
+          fullName: '',
+          avatar:
+              'https://img.hoidap247.com/picture/question/20200718/large_1595063159202.jpg',
+          email: '');
```

Comment:
I see you use mock data like this multiple time, why don't you create a mock class and just call?

Reply:
That's a good idea. I fixed it